test/integration/model_photo_shoot.cc

```diff
@@ -0,0 +1,265 @@
+          getline( iss, word, ' ' );
+          this->scaling = std::stod(word);
+          break;
+        }
+      }
+    }
+    poseFile.close();
+    EXPECT_EQ(remove(_poseFile.c_str()), 0);
+  }
+
+  // Documentation inherited
+  protected: void SetUp() override
+  {
+    InternalFixture::SetUp();
+
+    auto plugin = sm.LoadPlugin("libMockSystem.so",
+                                "ignition::gazebo::MockSystem",
+                                nullptr);
+    EXPECT_TRUE(plugin.has_value());
+    this->systemPtr = plugin.value();
+    this->mockSystem = static_cast<gazebo::MockSystem *>(
+        systemPtr->QueryInterface<gazebo::System>());
+  }
+
+  /// \brief Tests the Model Photo Shoot plugin with a given sdf world.
+  /// \param[in] _sdfWorld SDF World to use for the test.
+  protected: void modelPhotoShootTestCmd(const std::string _sdfWorld)
+  {
+    // First run of the server generating images through the plugin.
+    this->serverConfig.SetResourceCache(test::UniqueTestDirectoryEnv::Path());
+    this->serverConfig.SetSdfFile(
+        common::joinPaths(PROJECT_SOURCE_PATH, _sdfWorld));
+
+    this->server = std::make_unique<Server>(this->serverConfig);
+    EXPECT_FALSE(this->server->Running());
+    EXPECT_FALSE(*this->server->Running(0));
+
+    this->server->SetUpdatePeriod(1ns);
+    this->server->Run(true, 50, false);
+
+    common::ConnectionPtr postRenderConn;
+
+    // A pointer to the ecm. This will be valid once we run the mock system.
+    gazebo::EntityComponentManager *ecm = nullptr;
+    this->mockSystem->preUpdateCallback =
+      [&ecm](const gazebo::UpdateInfo &, gazebo::EntityComponentManager &_ecm)
+      {
+        ecm = &_ecm;
+      };
+    this->mockSystem->configureCallback =
+      [&](const gazebo::Entity &,
+            const std::shared_ptr<const sdf::Element> &,
+            gazebo::EntityComponentManager &,
+            gazebo::EventManager &_eventMgr)
+      {
+        postRenderConn = _eventMgr.Connect<gazebo::events::PostRender>(
+            std::bind(&ModelPhotoShootTest::OnPostRender, this));
+      };
+    this->LoadPoseValues();
+    this->server->AddSystem(this->systemPtr);
+
+    while (!testPicturesTaken)
+    {
+      this->server->Run(true, 1, false);
+      std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    }
+    testImages("1.png", "1_test.png");
+    testImages("2.png", "2_test.png");
+    testImages("3.png", "3_test.png");
+    testImages("4.png", "4_test.png");
+    testImages("5.png", "5_test.png");
+  }
+
+  public: ServerConfig serverConfig;
+  public: std::unique_ptr<Server> server;
+  public: ignition::gazebo::SystemPluginPtr systemPtr;
+  public: gazebo::MockSystem *mockSystem;
+
+  private: gazebo::SystemLoader sm;
+  private: bool testPicturesTaken{false};
+  private: double scaling;
+  private: ignition::math::Vector3d translation;
+};
+
+// Test the Model Photo Shoot plugin on the example world.
+TEST_F(ModelPhotoShootTest, ModelPhotoShootExampleWorld)
```

Comment:
I've created an issue about not being able to have two tests here (https://github.com/ignitionrobotics/ign-gazebo/issues/1410). I think the easiest workaround is to have two separate test files and share the test code between them.

Reply:
Yup, done that, thanks for the issue.